examples/ansible/setup_libvirt_host.yml

```
@@ -25,36 +29,35 @@
         name: libvirtd
         wait_for:
           path: /var/run/libvirt/libvirt-sock
-
   tasks:
-
     - name: Retrieve image for workload {{ workload.name }}
       become: true
       containers.podman.podman_image:
         name: "{{ workload.image }}"
         state: present
       register: retrieve_workload_image
+      failed_when: retrieve_workload_image is failed
 
     - name: Install tools for workload {{ workload.name }}
       become: true
       ansible.builtin.command: >-
         podman container runlabel {{ workload.labels.install }} {{ workload.image }}
       register: workload_runlabel_install
-      changed_when:
-        - ('already exist' not in workload_runlabel_install.stdout)
+      changed_when: "'already exist' not in workload_runlabel_install.stdout"
+      failed_when: workload_runlabel_install is failed
 
     - name: Install required packages, if any, for workload {{ workload.name }}
       become: true
       ansible.builtin.package:
         name: "{{ workload.required_pkgs }}"
         state: present
       register: required_pkgs_install
-      when:
-        - ((workload.required_pkgs | default([])) | length) > 0
+      when: ((workload.required_pkgs | default([])) | length) > 0
+      failed_when: required_pkgs_install is failed
+
 
-    - name: Handle reboot, if needed
-      when:
-        - required_pkgs_install is changed
+    - name: Handle manual reboot, if needed
+      when: required_pkgs_install.changed and (reboot_mode | default('manual')) == 'manual'
```

Note that the `*when` condition list syntax avoids the need to explicitly use the `and` keyword when adding additional conditions to an exiting condition list, as the conditions specified in the list are implicitly "anded" together.

This leads to easier to read code and eliminates the need for changes to existing lines in such cases.

```suggestion
      when:
        - required_pkgs_install.changed
        - (reboot_mode | default('manual')) == 'manual'
```